packages/react-tooltip/src/components/Tooltip/renderTooltip.tsx

```diff
@@ -11,9 +12,16 @@ export const renderTooltip = (state: TooltipState) => {
   const { slots, slotProps } = getSlots(state, tooltipShorthandProps);
 
   return (
-    <slots.root {...slotProps.root}>
-      <slots.arrow {...slotProps.arrow} />
+    <>
       {state.children}
-    </slots.root>
+      {state.shouldRenderTooltip && (
+        <Portal>
+          <slots.root {...slotProps.root}>
+            {!state.noArrow && <div ref={state.arrowRef} className={state.arrowClassName} />}
```

Comment:
The arrow slot is not part of the public API, and making it a slot added extra overhead (e.g. using a merged ref for the arrow slot, even though it would never have an external ref), and functionality that would never be used, which potentially increases maintenance cost. The only thing the arrow needs is a class and a ref for popper, so I only included the minimum necessary.